### PR TITLE
Fix Field3D::setBoundaryTo for FCI methods

### PR DESCRIPTION
**Pull request by bendudson**
Without this fix, boundary conditions set on yup/down fields are not applied when a boundary is copied from one field to another.

@dschwoerer This is needed in Hermes-3 with FCI so that this line correctly applies the sheath boundary conditions to the density field:
https://github.com/bendudson/hermes-3/blob/master/src/evolve_density.cxx#L217

### CODE REVIEW

File: src/field/field3d.cxx [hunk 1]
```diff
@@ -491,7 +491,25 @@ void Field3D::setBoundaryTo(const Field3D& f3d) {
...
+    // Set yup/ydown using midpoint values from f3d
+    ASSERT1(f3d.hasParallelSlices());
+    ASSERT1(hasParallelSlices());
+
+    for (auto& region : fieldmesh->getBoundariesPar()) {
+      for (const auto& pnt : *region) {
+        // Interpolate midpoint value in f3d
+        const BoutReal val = pnt.interpolate_sheath_o1(f3d);
+        // Set the same boundary value in this field
+        pnt.dirichlet_o2(*this, val);
```
Comment (dschwoerer) [inline src/field/field3d.cxx:504]: We should either take o2 twice, or o1 twice:
```suggestion
        pnt.dirichlet_o1(*this, val);
```

I think `o1` should be sufficient, as `f3d` and `*this` should be close. But I am not sure what is more stable ...

File: src/field/field3d.cxx [hunk 2]
```diff
@@ -491,7 +491,25 @@ void Field3D::setBoundaryTo(const Field3D& f3d) {
...
+      }
+    }
+    return;
```
Comment (dschwoerer) [inline src/field/field3d.cxx:507]: For hermes that might work, but should we not also copy the x-boundary conditions with FCI?
```suggestion
```

**Review by dschwoerer — CHANGES_REQUESTED** [review]

File: src/field/field3d.cxx (hunk 1 above)
Comment (dschwoerer) [reply src/field/field3d.cxx:504]: The FA code also uses first order, so I have pushed a fixed based on that ...

**Comment by dschwoerer** [comment]
@bendudson Are you ok with these changes?

**Comment by dschwoerer** [comment]
I just noticed #2962 is also open - is the other still needed with this one?

**Review by bendudson — COMMENTED** [review]
Thanks @dschwoerer ! The changes look good to me.
I think #2962 is not needed once this is merged.

**Review by dschwoerer — APPROVED** [review]

**Pull request merged**